node/src/service.rs

```diff
@@ -238,23 +265,34 @@ async fn build_relay_chain_interface(
 	}
 }
 
-// We use this as the RuntimeApi definition in the following function so that we don't need to specify so many generic traits.
-// This might have to change if we want to use this function for other runtimes.
-use amplitude_runtime::RuntimeApi as AmplitudeRuntimeApi;
+type FullDepsOf<RuntimeApi, Executor> = FullDeps<
+	TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>,
+	sc_transaction_pool::FullPool<
+		Block,
+		TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>,
+	>,
+>;
 
-/// Start a node that exposes the RPC endpoints for the spacewalk pallets for the amplitude runtime.
 #[sc_tracing::logging::prefix_logs_with("Parachain")]
-async fn start_node_impl_amplitude<Executor>(
+async fn start_node_impl<RuntimeApi, Executor>(
```

Comment:
All the four `start_node_impl_<...>` functions were *exactly* the same with the exception being a single line that calls either one of the four functions in `rpc.rs`. I now added the respective function as an extra argument and define which of the functions to use when calling `start_node_impl` in any one of the four `start_parachain_node_<...>` functions at the end of this file.